### PR TITLE
Do not rely on string comparison in truncindex

### PR DESCRIPTION
**Pull request by LK4D4**

### CODE REVIEW

File: pkg/truncindex/truncindex.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ var (
 
 	// ErrIllegalChar is returned when a space is in the ID
 	ErrIllegalChar = errors.New("illegal character: ' '")
+
+	// ErrIsNotExist is returned when ID or its prefix not found in index.
```
Comment (tonistiigi) [inline pkg/truncindex/truncindex.go]: s/ErrIsNotExist/ErrNotExist/

**Comment by LK4D4** [comment]
@tonistiigi fixed, thanks!

**Comment by tonistiigi** [comment]
LGTM

**Comment by jfrazelle** [comment]
nice LGTM

**Comment by stevvooe** [comment]
LGTM

**Comment by tonistiigi** [comment]
@LK4D4 Tests caught to `id` to `ID` change.

**Comment by stevvooe** [comment]
@LK4D4 I hope you learned your lesson. :stuck_out_tongue:

**Comment by LK4D4** [comment]
@tonistiigi @stevvooe I think I got it this time :)

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**